caas/kubernetes/provider/credentials.go

```diff
@@ -40,7 +42,15 @@ func (environProviderCredentials) DetectCredentials() (*cloud.CloudCredential, e
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	caasConfig, err := clientConfigFunc()
+
+	kubeConfigPath := clientconfig.GetKubeConfigPath()
```

Comment:
This doesn't belong here. 
NewClientConfigReader("kubernetes") above is the function that takes a caas type ("kubernetes" currently) and returns a func to get a client config. Any substrate specific code belongs in there. This detect credentials method should know nothing about the internals of how the client config works.